spec/02-integration/05-proxy/01-router_spec.lua

```diff
@@ -149,6 +149,58 @@ describe("Router", function()
     end)
   end)
 
+  describe("URI arguments (querystring)", function()
+
+    setup(function()
+      insert_apis {
+        {
+          name = "api-1",
+          upstream_url = "http://httpbin.org",
+          hosts = { "example.com" },
+        },
+      }
+
+      assert(helpers.start_kong())
+    end)
+
+    teardown(function()
+      helpers.stop_kong()
+    end)
+
+    it("preserves URI arguments", function()
+      local res = assert(client:send {
+        method = "GET",
+        path   = "/get",
+        query  = {
+          foo   = "bar",
+          hello = "world",
+        },
+        headers = {
+          ["Host"] = "example.com",
+        },
+      })
+
+      local body = assert.res_status(200, res)
+      local json = cjson.decode(body)
+      assert.equal("bar", json.args.foo)
+      assert.equal("world", json.args.hello)
+    end)
+
+    it("does not proxy '$is_args' if URI does not contain arguments", function()
```

Comment:
@bungle under "Normalizations that change semantics"

```
Removing the "?" when the query is empty. When the query is empty, there may be no need for the "?". 
Example:
http://www.example.com/display? → http://www.example.com/display
```

The point here is that we want to avoid changing semantic.